src/base-traits.jl

```diff
@@ -6,6 +6,10 @@ using Compat
 export IsLeafType, IsBits, IsImmutable, IsContiguous, IsIndexLinear,
        IsAnything, IsNothing, IsCallable
 
+@static if VERSION >= v"0.6.0-dev"
```

Comment:
BTW: My comment above about the new type system was related to a "complete" solution to this type of interface specification, as it is done in Traits.jl.